### PR TITLE
More complete & accurate keybinding editor documentation

### PR DESCRIPTION
**Pull request by michaelpacer**
This should address #2153, though there is still room for improvement. 

In particular, I would like to add a shorter default display, with a "see more" button that will expand into the current text. 

That feature is not a blocking feature though, as it is a inconvenience only if you need to scroll through it frequently (which I've only noticed because of the amount of time I spent testing the editor).

### CODE REVIEW

File: notebook/static/notebook/js/shortcuteditor.js [hunk 1]
```diff
@@ -106,27 +106,55 @@ var KeyBindingList = createClass({
...
+            "This dialog allows you to modify the keyboard shortcuts available in command mode. "+ 
+            "Any changes will be persisted between sessions and across environments. "+
+            "You can define two kinds of shorctuts: **key combinations** and **key sequences**.\n"+
+            "\n"+
+            " - **Key Combinations**:\n"+
+            "   - Use hyphens `-` to represent keys that should be pressed at the same time.\n"+
+            "   - This is designed for use with *modifier* keys: `Cmd`, `Ctrl`, `Alt` ,`Meta`, "+
+            "`Cmdtrl`, and `Shift`.\n"+ 
+            "   - At most, one non-modifier key can exist in a key combination.\n"+
+            "   - Multiple non-modifier key can exist in a key combination.\n"+
```
Comment (takluyver) [inline notebook/static/notebook/js/shortcuteditor.js]: This line appears to directly contradict the one above it. Did you mean to remove 'non-'?

**Comment by takluyver** [comment]
This is great, but:

- We're still missing an explanation of 'Cmdtrl'
- I am a bit concerned that is now so long that I don't see the actual shortcut editing area without scrolling down. Maybe an easy way to get round this for 5.0 would be to move the explanation to the end of the dialog, and have just the first bit at the top, like:

> Here you can modify the keyboard shortcuts available in command mode. Your changes will be stored for later sessions. See more **details of defining keyboard shortcuts** below.

(with a link from the bold text to scroll the help info into view)

File: notebook/static/notebook/js/shortcuteditor.js (hunk 1 above)
Comment (michaelpacer) [inline notebook/static/notebook/js/shortcuteditor.js]: Yes!

**Comment by michaelpacer** [comment]
I like that idea, but how should I handle the text scrolling? Just an anchor tag with an id inside the beginning of the detailed explanation and a hyperlink?

**Comment by takluyver** [comment]
I think that should work. Try it and see :-)

File: notebook/static/notebook/js/shortcuteditor.js [hunk 2]
```diff
@@ -106,27 +106,55 @@ var KeyBindingList = createClass({
...
+            "   - *Invalid Example*s: `a-b` and `a-Ctrl-Shift`. \n"+
+            " - **Key Sequences**:\n"+
+            "   - Use commas `,` to represent keys that should be pressed in sequence.\n"+
+            "   - The order in which keys must be pressed exactly matches the left-to-right order of "+
+            "the characters in the sequence, with no interruptions.\n"+
+            "     - E.g., `h,a,l,t` would be triggered by typing <kbd>h</kbd> <kbd>a</kbd> "+
+            "<kbd>l</kbd> <kbd>t</kbd> but not <kbd>h</kbd> <kbd>a</kbd> <kbd>a</kbd> <kbd>l</kbd> "+
+            "<kbd>t</kbd> or <kbd>a</kbd>  <kbd>h</kbd> <kbd>l</kbd> <kbd>t</kbd>.\n"+
+            "   - Sequences can include the same key multiple times (e.g., `d,d`).\n"+
+            "   - You cannot include a sequence that is a 'prefix' of another sequence.\n"+
```
Comment (Carreau) [inline notebook/static/notebook/js/shortcuteditor.js]: I read the the opposite way: `if d,d,d is bound you cannot bind d,d`. Both are true.

**Comment by takluyver** [comment]
Nice, thanks Mike. :-)

**Pull request merged**